comp/snmptraps/server/impl/server.go

```diff
@@ -0,0 +1,98 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2020-present Datadog, Inc.
+
+// Package serverimpl implements the traps server.
+package serverimpl
+
+import (
+	"context"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/hostname"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	trapsconf "github.com/DataDog/datadog-agent/comp/snmptraps/config"
+	"github.com/DataDog/datadog-agent/comp/snmptraps/forwarder"
+	"github.com/DataDog/datadog-agent/comp/snmptraps/listener"
+	"github.com/DataDog/datadog-agent/comp/snmptraps/server"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+)
+
+// Module defines the fx options for this component.
+var Module = fxutil.Component(
+	fx.Provide(newServer),
+)
+
+// Server manages an SNMP trap listener.
+type Server struct {
+	Addr      string
+	config    *trapsconf.TrapsConfig
+	listener  listener.Component
+	forwarder forwarder.Component
+	logger    log.Component
+}
+
+type dependencies struct {
+	fx.In
+	Config    trapsconf.Component
+	Listener  listener.Component
+	Logger    log.Component
+	Sender    sender.Sender
+	Forwarder forwarder.Component
+	Hostname  hostname.Component
+}
+
+// newServer configures a netflow server.
+func newServer(lc fx.Lifecycle, deps dependencies) server.Component {
+	conf := deps.Config.Get()
+
+	server := &Server{
+		config:    conf,
+		listener:  deps.Listener,
+		forwarder: deps.Forwarder,
+		logger:    deps.Logger,
+	}
+	if conf.Enabled {
+		lc.Append(fx.Hook{
+			OnStart: func(ctx context.Context) error {
+				return server.Start()
+			},
+			OnStop: func(ctx context.Context) error {
+				server.Stop()
+				return nil
+			},
+		})
+	}
+	return server
+}
+
+// Start starts the server listening
+func (s *Server) Start() error {
```

Comment:
This method doesn't have to be exported, ie, `start`.

Reply:
Done. I actually removed this entirely - the server doesn't do anything now that the forwarder and listener use the lifecycle hooks, it now only exists as something you can inject that will make both the forwarder and the listener start running. But I also made the Start/Stop methods unexported on the forwarder and listener.